xml_converter/src/xml_converter.cpp

```diff
@@ -327,13 +485,19 @@ int main() {
     map<string, Category> marker_categories;
     test_proto();
 
+    if (mkdir("./protobins/Data", 0700) == -1 && errno != EEXIST) {
+        cout << "Error making ./protobins/Data" << endl;
+        throw std::error_code();
+    }
+
     string directory = "./packs";
     DIR* dir = opendir(directory.c_str());
     struct dirent* entry = readdir(dir);
     while ((entry = readdir(dir)) != NULL) {
-        if (entry->d_type == DT_DIR) {
+        if (entry->d_type == DT_DIR && strcmp(entry->d_name, ".")) {
```

Comment:
As discussed, `strcmp()` returns a 0 when the strings are identical

Reply:
This is really a `strcmp` issue but I think there should be some additional stuff here. Maybe in `string_helper` (which really is doing stuff on std::string but maybe cstrings are close enough) you can write a `streq()` function that just does something like `bool cstring_equals(char*a, char*b){return strcmp(a,b) == 0;}` with a comment explaining what that function does and maybe why.

If this were solely a cpp repo we used `strcmp` a bunch all over the repo it would probably be fine, I would still want to change it though. But because even this individual repo is pretty polyglot (gdscript, rust, python, c, c++) (wow that's a lot) we should try to make code as readable as possible without knowing specific function detials, such as "strcmp returns an equivalent to `false` value of `0` if the strings are equal"

Reply:
`has_suffix()` check for suffixes. This would mean that we would also flag any folder that had "." as the final character. Additionally `has_suffix()` operates on std::string not `cstring`s (`char*` s), this means that ever time the function is called with char* a new std::string has to be allocated and constructed for it. Not a huge performance hit, especially for us, but another reason not to use `has_suffix()`. Maybe we want to consider moving all of the string_helper functions to use something called "string views" instead. And now that you point it out, would `has_suffix()` be better suited within `string_helper.cpp` instead?

Reply:
Good point. I changed it to explicity look for `"."` or `".."`

I like the idea of moving the function to string_helper. I added another function there that checks for the directories of a file path as well.